cubi-tools/prototypes/update_workflow.py

```diff
@@ -0,0 +1,646 @@
+                    f"'{branch_workflow_version}'!"
+                )
+            else:
+                answer_is_pos = user_response(
+                    "\nYou updated your local repo with the 'template-snakemake' "
+                    f"in branch/version tag '{branch}'."
+                    "\nDo you want to update the workflow version in "
+                    "'pyproject.toml'"
+                )
+
+                if answer_is_pos:
+                    with open(
+                        pathlib.Path(workflow_target, "pyproject.toml"),
+                        "w",
+                        encoding="utf-8",
+                    ) as text_file:
+                        text_file.write(toml.dumps(target_pyproject, encoder=None))
+                    print(
+                        "\nMetadata version in 'pyproject.toml' was updated from "
+                        f"version '{target_metadata_version}' to version "
+                        f"'{branch_metadata_version}'! and "
+                        "\nWorkflow version in 'pyproject.toml' was updated from "
+                        f"version '{target_workflow_version}' to version "
+                        f"'{branch_workflow_version}'!"
+                    )
+                else:
+                    print(
+                        "The 'pyproject.toml' metadata version was NOT updated from "
+                        f"version '{target_metadata_version}' to version "
+                        f"'{branch_metadata_version}' and "
+                        "The 'pyproject.toml' workflow version was NOT updated from "
+                        f"version '{target_workflow_version}' to version "
+                        f"'{branch_workflow_version}'!"
+                    )
+        else:
+            print(
+                "\nMetadata and workflow versions in 'pyproject.toml' are "
+                "up-to-date!\n"
+            )
+    return None
+
+
+def user_response(question, attempt=0):
+    """
+    Function to evaluate the user response to the Yes or No question refarding updating
+    the metadata files.
+    """
+    attempt += 1
+    prompt = f"{question}? (y/n): "
+    answer = input(prompt).strip().lower()
+    pos = ["yes", "y", "yay"]
+    neg = ["no", "n", "nay"]
+    if attempt == 2:
+        print("YOU HAVE ONE LAST CHANCE TO ANSWER THIS (y/n) QUESTION!")
+    if attempt >= 3:
+        raise RuntimeError(
+            "I warned you! You failed 3 times to answer a simple (y/n)"
+            " question! Please start over!"
+        )
+    if not (answer in pos or answer in neg):
+        print(f"That was a yes or no question, but you answered: {answer}")
+        return user_response(question, attempt)
+    return answer in pos
+
+
+def find_all_subdir_in_branch(workflow_branch):
+    """
+    Function to list all subdirectories in the 'template_snakemake' repo
+    to be able to create all missing subfolders.
+    """
+    subfolders = [
+        branch_folder.path
```

Comment:
this looks like it could be simplified/made more direct by using the `pathlib.iterdir` functionality

Reply:
Done